src/main/java/org/springframework/data/hazelcast/repository/support/StringBasedHazelcastRepositoryQuery.java

```diff
@@ -40,11 +43,27 @@ public StringBasedHazelcastRepositoryQuery(HazelcastQueryMethod queryMethod, Haz
     @Override
     public Object execute(Object[] parameters) {
         String queryStringTemplate = queryMethod.getAnnotatedQuery();
-        String queryString = String.format(queryStringTemplate, parameters);
+        String queryString = String.format(queryStringTemplate, formatParameters(parameters));
         SqlPredicate sqlPredicate = new SqlPredicate(queryString);
         return getMap(keySpace).values(sqlPredicate);
     }
 
+    private Object[] formatParameters(Object[] parameters) {
```

Comment:
what happens if the parameter is a collection with non-primitive objects? something like
`public List<Person> peopleWithLastNameIn(Collection<SomeClass> lastNames);`
Do we throw UnsupportedException?

Reply:
If your non-primitive class implements `toString()` in a way that it makes sense for the query, then it's fine. Otherwise you'll get an exception from the Hazelcast Query mechanism, which is fine imo. We should not validate it at this point.